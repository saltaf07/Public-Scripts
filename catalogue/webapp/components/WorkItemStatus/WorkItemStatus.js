// @flow
import Space from '@weco/common/views/components/styled/Space';
import { type PhysicalItemWithStatus } from '@weco/common/utils/works';

type Props = {| item: PhysicalItemWithStatus |};

const WorkItemStatus = ({ item }: Props) => {
  return item.status ? (
    <Space
      v={{
        size: 'l',
        properties: ['margin-bottom'],
      }}
    >
      <p className="no-margin">
        <span data-test-id="itemStatus">{item.status.label}</span>
      </p>
    </Space>
  ) : null;
};

export default WorkItemStatus;
