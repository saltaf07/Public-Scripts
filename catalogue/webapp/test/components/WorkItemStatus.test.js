// @flow
import WorkItemStatus from '@weco/catalogue/components/WorkItemStatus/WorkItemStatus';
import {
  mountWithTheme,
  updateWrapperAsync,
} from '@weco/common/test/fixtures/enzyme-helpers';
// import mockCatalogueWork from '@weco/catalogue/__mocks__/catalogue-work';
import mockStacksWork from '@weco/catalogue/__mocks__/stacks-work';

const availableItem = mockStacksWork.items.find(
  item => item.status.label === 'Available'
);

describe('Feature: 1. As a library member I want to know if an item is available for requesting', () => {
  test.only(`
    Scenario: I'm viewing a work page
      Given the work has a requestable item
      And it's available to request
      Then I can see that it's requestable
  `, async () => {
    const wrapper = mountWithTheme(<WorkItemStatus item={availableItem} />);
    await updateWrapperAsync(wrapper);
    expect(wrapper.find('[data-test-id="itemStatus"]')).toMatchSnapshot();
  });

  test.todo(`
    Scenario: I'm viewing a work page
      Given the work has a requestable item
      And itâ€™s unavailable to request
      Then I can see a that it's temporarily unavailable
  `);

  test.todo(`
    Scenario: I'm viewing a work page
      Given the work has no requestable items
      Then I can see a that it's not requestable and why (closed or open shelves or unknown)
  `);
});
