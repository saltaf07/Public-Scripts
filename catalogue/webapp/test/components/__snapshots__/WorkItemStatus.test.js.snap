// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature: 1. As a library member I want to know if an item is available for requesting 
    Scenario: I'm viewing a work page
      Given the work has a requestable item
      And it's available to request
      Then I can see that it's requestable
   1`] = `
<span
  data-test-id="itemStatus"
>
  Available
</span>
`;

exports[`Feature: 1. As a library member I want to know if an item is available for requesting 
    Scenario: I'm viewing a work page
      Given the work has a requestable item
      And itâ€™s unavailable to request
      Then I can see a that it's temporarily unavailable
   1`] = `
<span
  data-test-id="itemStatus"
>
  On holdshelf
</span>
`;

exports[`Feature: 1. As a library member I want to know if an item is available for requesting 
    Scenario: I'm viewing a work page
      Given the work has no requestable items
      Then I can see a that it's not requestable and why (closed or open shelves or unknown)
   1`] = `
<span
  data-test-id="itemStatus"
>
  Missing
</span>
`;
